task/fbc-related-image-check/0.1/fbc-related-image-check.yaml

```
@@ -28,20 +28,23 @@ spec:
           cpu: 10m
       script: |
         #!/usr/bin/env bash
-        set -o pipefail
+        set -euo pipefail
         source /utils.sh
+        trap 'handle_error $(results.TEST_OUTPUT.path)' EXIT
+
         FAILEDIMAGES=""
         catalog="$(find $(workspaces.workspace.path)/hacbs/fbc-validation/ -name catalog.yaml)"
 
         # OPM generates catalog file in a way that yaml file could contain stream of JSON objects
         # thats why we need jq in for this situation, because yq can't parse this file
         # however there is also posibility that catalog.yaml has yaml data in it
 
-        relImgs="$(yq -r '.relatedImages[]?.image' $catalog)"
-        if [ $? -ne 0 ]; then
+        status=0
+        relImgs="$(yq -r '.relatedImages[]?.image' $catalog)" || status=$?
+        if [ $status -ne 0 ]; then
           echo "Processing the catalog with yq failed because catalog.yaml contains data type other than yaml. Attempting to process with jq..."
-          relImgs="$(jq -r '.relatedImages[]?.image' $catalog)"
-          if [ $? -ne 0 ]; then
+          relImgs="$(jq -r '.relatedImages[]?.image' $catalog)" || status=$?
```

Status must reset to 0 manually, `|| status=$?` is executed only when previous command failed, thus it would never reset it back to 0 and next condition would fail
```suggestion
          status=0
          relImgs="$(jq -r '.relatedImages[]?.image' $catalog)" || status=$?
```

---

Yes, you are right. Updated. Thanks!